### PR TITLE
#3968, Ambiguous description of COM_STMT_EXECUTE in MySQL protocol

### LINKED ISSUE #3968: Ambiguous description of COM_STMT_EXECUTE in MySQL protocol
The description of COM_STMT_EXECUTE command like this:

image

What if the **num-params** equals 0? 

Implementation of Connector/J like this:

image

While Implementation of sysbench like this:

image

The former has 1 more byte than latter:

image

Sharding-Proxy's command handler only support Connector/J's implementation for now, so access by sysbench will be failed. We need to support all two implementations.

### PR DESCRIPTION
**Pull request by tuohai666**
Fixes #3968.

### CODE REVIEW

File: sharding-proxy/sharding-proxy-transport/sharding-proxy-transport-mysql/src/main/java/org/apache/shardingsphere/shardingproxy/transport/mysql/packet/command/query/binary/execute/MySQLComStmtExecutePacket.java [hunk 1]
```diff
@@ -86,7 +86,7 @@ public MySQLComStmtExecutePacket(final MySQLPacketPayload payload) throws SQLExc
         } else {
             nullBitmap = null;
             newParametersBoundFlag = null;
-            parameters = null;
+            parameters = new ArrayList<>(0);;
```
Comment (terrymanu) [inline sharding-proxy/sharding-proxy-transport/sharding-proxy-transport-mysql/src/main/java/org/apache/shardingsphere/shardingproxy/transport/mysql/packet/command/query/binary/execute/MySQLComStmtExecutePacket.java]: How about use Collections.emptyList()?

**Comment by terrymanu** [comment]
Please set milestone and lable

File: sharding-proxy/sharding-proxy-transport/sharding-proxy-transport-mysql/src/main/java/org/apache/shardingsphere/shardingproxy/transport/mysql/packet/command/query/binary/execute/MySQLComStmtExecutePacket.java (hunk 1 above)
Comment (tuohai666) [reply sharding-proxy/sharding-proxy-transport/sharding-proxy-transport-mysql/src/main/java/org/apache/shardingsphere/shardingproxy/transport/mysql/packet/command/query/binary/execute/MySQLComStmtExecutePacket.java]: Yes, it's better.

**Pull request merged**